### ISSUE TITLE
GCC Color Diagnostics

### ISSUE THREAD

**Issue by RobertBColton** [appveyor/ci#1849]
So GCC added native color diagnostics in 4.9 that defaults to enabled when the output is a terminal. The output is encoded using ANSI escapes. Because the GCC in the mingw-w64 installed by default in AppVeyor is 6.3 I thought that make should be showing color diagnostics for warnings and errors, but this is not the case. Currently my same build with GCC shows color in Travis if I upgrade it to use GCC 7.* or if I use Clang.

https://gcc.gnu.org/onlinedocs/gcc-7.2.0/gcc/Diagnostic-Message-Formatting-Options.html

I've experimented with the following in my ``appveyor.yml`` file:
```yml
  - bash -lc "export GCC_COLORS='error=01;31:warning=01;35:note=01;36:caret=01;32:locus=01:quote=01'"
  - set GCC_COLORS="error=01;31:warning=01;35:note=01;36:caret=01;32:locus=01:quote=01"
  - setx GCC_COLORS "error=01;31:warning=01;35:note=01;36:caret=01;32:locus=01:quote=01"
```

I would appreciate any insight into how to get color diagnostics enabled in my AppVeyor project.

**Comment by FeodorFitsner** [comment]
Can you drop a link to your test build please?

**Comment by RobertBColton** [comment]
Yes certainly, I've been changing some things to make better use of the cache. But any of the passing jobs here demonstrate it:
https://ci.appveyor.com/project/enigma-dev/enigma-dev/history

Including the following:
https://ci.appveyor.com/project/enigma-dev/enigma-dev/build/1.0.93/job/4x30sp04dxp9dmgf

**Comment by RobertBColton** [comment]
It would be nice if this fix could be applied in AppVeyor:
https://stackoverflow.com/a/2095623

**Comment by XVilka** [comment]
Would be nice to revisit this issue - it would help to spot the problems in lengthy logs easier.
